### ISSUE TITLE
[datalake-store-filesystem] MkdirsResponder tries to parse non-existing error response.

### ISSUE THREAD

**Issue by afsalthaj** [Azure/azure-sdk-for-go#901]
Currently, MkdirResponder doesn't work as it uses the wrong status code when trying to create a `RespondDecorator`. It checks for status code 200 in the response (while it is 201 Accepted), and tries to parse the error response and fails to do it. We get the following error:

 "autorest/azure: error response cannot be parsed".

Fix:  Pass `http.StatusCreated` instead of `http.StatusOk` to `WithErrorUnlessStatusCode` in the function`MkdirsResponder`.

**Comment by joshgav** [comment]
Thanks @afsalthaj. Can you share the line of code which leads to this error?

This problem might go back to the Swagger spec here: https://github.com/Azure/azure-rest-api-specs/blob/current/specification/datalake-store/data-plane/Microsoft.DataLakeStore/2016-11-01/filesystem.json#L302

cc @begoldsm

**Comment by afsalthaj** [comment]
@joshgav

```
var tooId bool = false //get from somewhere else

	acl, err := datalakeFsClient.GetACLStatus(accountName, pathParameter, "GETACLSTATUS", &tooId)
```

Thanks

[I have raised an issue on avoid passing  operation constants (ex: GETACLSTATUS)  from user side, #905]

**Comment by afsalthaj** [comment]
@joshgav  Thanks for getting back. I think I got the issue resolved by passing the actual operation constant which is MKDIRS. The MkDirs was creating a file instead of a directory since I was passing "CREATE" as operation constant.

Meanwhile, I have created this issue #905 to avoid issues like these.

**Comment by begoldsm** [comment]
@afsalthaj is correct. CREATE is only for the creation of files, and does return a 201 on success (and the swagger reflects this). For directory creation, the MKDIRS op code must be used, which does properly return a 200 on success, this is not a spec bug. In general, GO should follow the practice for other languages, which is that enums with only a single value should be treated as constants and not as params to the functions, and explicitly injected into the request instead of allowing for a user to change them. See this generated code in C# for example:
https://github.com/Azure/azure-sdk-for-net/blob/psSdkJson6/src/SDKs/DataLake.Store/Management.DataLake.Store/Generated/FileSystemOperations.cs#L732
Notice that the method does not take in an operation parameter, it instead sets it internally in the method.

**Comment by jhendrixMSFT** [comment]
This has been fixed in v14.0.0, we now pass constants as described above.  If this is still an issue with the latest SDK please re-open and we can investigate further.
